### PR TITLE
Add a step that checks that all steps have been executed

### PR DESCRIPTION
**Pull request by ujappelbe**

### CODE REVIEW

File: parseresults.py [hunk 1]
```diff
@@ -29,10 +29,14 @@
 with open(inputFile, "r") as myfile:
     data = myfile.read()
 
-result = TestResult(data);
+numTests = -1
+reNumTests = re.compile(r'INSTRUMENTATION_STATUS: numtests=(\d*)$')
+for line in data.split('\n'):
+ if(reNumTests.match(line)):
+     numTests = reNumTests.search(line).group(1)
```
Comment (antnsi) [inline parseresults.py:36]: should break loop with the first catch, or it will overwrite previous value

File: parseresults.py [hunk 2]
```diff
@@ -43,8 +47,7 @@
         else:
             failures += 1
 
-
-print("Amount of steps = " + str(len(testResults)))
+print("Amount of steps = " + str(len(testResults)) + " (Claimed " + numTests + ")" )
```
Comment (antnsi) [inline parseresults.py:54]: I prefer to use print "text %i text".format(int), than concatenation :)

File: parseresults.py (hunk 1 above)
Comment (ujappelbe) [inline parseresults.py:36]: Added break as suggested :+1:

File: parseresults.py (hunk 2 above)
Comment (ujappelbe) [inline parseresults.py:54]: Like this?
`print("Amount of steps = {} (Claimed {})".format(len(testResults), numTests))`

It's a bit closer to c-style, I like the fact that it allows you to easier see how the string itself will look. But then seeing which variable occupies which space is tricky if there are several of them.

Or like this?
`print("Amount of steps = %i (Claimed %s)" % (len(testResults), numTests))`
it says here that it is depreciated in python 3.1 http://www.pythonforbeginners.com/concatenation/string-concatenation-and-formatting-in-python so I'd rather not use it.

File: parseresults.py [hunk 3]
```diff
@@ -42,14 +47,18 @@
             skipped += 1
         else:
             failures += 1
+if(numTests != str(len(testResults))):
+    failures += 1
 
+print("Amount of steps = " + str(len(testResults)) + " (Claimed " + numTests + ")" )
```
Comment (rasmus-unity) [inline parseresults.py:53]: Minor: "Expected" instead of "Claimed"?

**Pull request merged**